internal/controller/apiextensions/usage/reconciler.go

```diff
@@ -316,6 +317,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			}
 		}
 
+		if u.Spec.ReplayDeletion != nil && *u.Spec.ReplayDeletion && used.GetAnnotations() != nil && used.GetAnnotations()[usage.AnnotationKeyFirstDeletionAttempt] != "" {
+			// We have already recorded a deletion attempt and want to replay deletion, let's delete the used resource.
+			go func() {
+				// We do the deletion async and after some delay to make sure the usage is deleted before the
+				// deletion attempt. We remove the finalizer on this Usage right below, so, we know it will disappear
+				// very soon.
+				time.Sleep(2 * time.Second)
```

Comment:
This is the piece that I don't like most (which is why the PR is draft).
But without doing async with some delay, it becomes a chicken and egg problem, i.e. as long as the Usage is there, deletions will be blocked. /cc @bobh66

Reply:
I agree it's not ideal, but this is the same solution I ended up using for the python implementation which has been running for over a year in our deployments.